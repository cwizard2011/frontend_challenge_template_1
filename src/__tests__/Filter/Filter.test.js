import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Filter from '../../Filters';

describe('<Filter />', () => {
  const props = {
    filterCategory: jest.fn(),
    setDepartmentFilter: jest.fn()
  }
  it('renders without crashing', () => {
    const wrapper = shallow(<Filter />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('should toggle the category and set the value in state', () => {
    const wrapper = shallow(<Filter {...props} />);
    wrapper.instance().toggleCategory('Animal');
    expect(wrapper.state('collapseTextCategories')).toBe('Animal');
  });

  it('should toggle the department and set the value in state', () => {
    const wrapper = shallow(<Filter {...props} />);
    wrapper.instance().toggleDepartment('Nature');
    expect(wrapper.state('collapseTextDepartment')).toBe('Nature');
  });

  it('should call the toggle() method', () => {
    const wrapper = shallow(<Filter {...props} />);
    wrapper.instance().toggle('category');
    expect(wrapper.state('collapseCategory')).toEqual(true);
    expect(wrapper.state('collapseDepartment')).toEqual(false);

    wrapper.instance().toggle('department');
    expect(wrapper.state('collapseCategory')).toEqual(true);
    expect(wrapper.state('collapseDepartment')).toEqual(true);
  })
})