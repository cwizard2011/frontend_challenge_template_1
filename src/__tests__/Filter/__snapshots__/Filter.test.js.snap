// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filter /> renders without crashing 1`] = `
<div
  className="filter-cards-container"
>
  <div>
    <p
      className="filter-cards-button"
      onClick={[Function]}
    >
      Filter by Category
    </p>
    <Collapse
      appear={false}
      enter={true}
      exit={true}
      in={false}
      isOpen={true}
      mountOnEnter={false}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      tag="div"
      timeout={350}
      unmountOnExit={false}
    >
      <Form
        tag="form"
      >
        <FormGroup
          tag="fieldset"
        >
          <RadioButton
            categoryText="Filter by Categories"
            index={0}
            labelText="French"
            name="category"
            toggleClick={[Function]}
          />
          <RadioButton
            categoryText="Filter by Categories"
            index={1}
            labelText="Italian"
            name="category"
            toggleClick={[Function]}
          />
          <RadioButton
            categoryText="Filter by Categories"
            index={2}
            labelText="Irish"
            name="category"
            toggleClick={[Function]}
          />
          <RadioButton
            categoryText="Filter by Categories"
            index={3}
            labelText="Animal"
            name="category"
            toggleClick={[Function]}
          />
          <RadioButton
            categoryText="Filter by Categories"
            index={4}
            labelText="Flower"
            name="category"
            toggleClick={[Function]}
          />
          <RadioButton
            categoryText="Filter by Categories"
            index={5}
            labelText="Christmas"
            name="category"
            toggleClick={[Function]}
          />
          <RadioButton
            categoryText="Filter by Categories"
            index={6}
            labelText="Valentine's"
            name="category"
            toggleClick={[Function]}
          />
        </FormGroup>
      </Form>
    </Collapse>
  </div>
  <div>
    <p
      className="filter-cards-button"
      onClick={[Function]}
    >
      Filter by Department
    </p>
    <Collapse
      appear={false}
      enter={true}
      exit={true}
      in={false}
      isOpen={true}
      mountOnEnter={false}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      tag="div"
      timeout={350}
      unmountOnExit={false}
    >
      <Form
        tag="form"
      >
        <FormGroup
          tag="fieldset"
        >
          <RadioButton
            departmentText="Filter by Department"
            index={0}
            labelText="Regional"
            name="department"
            toggleClick={[Function]}
          />
          <RadioButton
            departmentText="Filter by Department"
            index={1}
            labelText="Nature"
            name="department"
            toggleClick={[Function]}
          />
          <RadioButton
            departmentText="Filter by Department"
            index={2}
            labelText="Seasonal"
            name="department"
            toggleClick={[Function]}
          />
        </FormGroup>
      </Form>
    </Collapse>
  </div>
</div>
`;
